sdks/java/core/src/main/java/org/apache/beam/sdk/testing/BigqueryMatcher.java

```diff
@@ -165,14 +165,16 @@ QueryResponse queryWithRetries(Bigquery bigqueryClient, QueryRequest queryConten
         }
       } catch (IOException e) {
         // ignore and retry
-        LOG.warn("Ignore the error and retry the query.");
+        LOG.warn("Retrying query ({}) after exception", queryContent.getQuery(), e);
```

Comment:
Also not accurate. You need to store this as `lastException` and log whenever you enter the loop and `lastException` is not null.